src/js/stores/DCOSStore.js

```diff
@@ -402,6 +409,23 @@ class DCOSStore extends EventEmitter {
    * @type {ServiceTree}
    */
   get serviceTree() {
+    if (!this._serviceTree) {
+      this.serviceTree = this.buildServiceTree();
+    }
+
+    return this._serviceTree;
+  }
+
+  set serviceTree(newValue) {
+    this._serviceTree = newValue;
+    this._flatServiceTree = null;
```

Comment:
It is a little weird for me that this is the only reference for the flat service tree. Is it missing a rebase, or is it lacking a  getter for the flatServiceTree?